tools/clang/unittests/HLSLExec/LongVectors.h

```diff
@@ -293,108 +456,210 @@ template <typename T> struct LongVectorOpTestConfig {
+      VERIFY_FAIL("Clamp is not supported for bools.");
+    else
+      return GetInputValueSetByKey<T>(InputArgsArrayName);
+
+    VERIFY_FAIL("Invalid type for args array.");
+    return std::vector<T>();
+  }
+
+  LongVectorBinaryOpType GetBinaryOpType() const { return BinaryOpType_; }
+
+  LongVectorUnaryOpType GetUnaryOpType() const { return UnaryOpType_; }
+
+  float GetTolerance() const { return Tolerance_; }
+
+  std::string GetCompilerOptionsString(size_t VectorSize) {
+    std::stringstream CompilerOptions("");
+    std::string HLSLType = GetHLSLTypeString();
+    CompilerOptions << "-DTYPE=";
+    CompilerOptions << HLSLType;
+    CompilerOptions << " -DNUM=";
+    CompilerOptions << VectorSize;
+    const bool Is16BitType =
+        (HLSLType == "int16_t" || HLSLType == "uint16_t" || HLSLType == "half");
+    CompilerOptions << (Is16BitType ? " -enable-16bit-types" : "");
+    CompilerOptions << " -DOPERATOR=";
+    CompilerOptions << OperatorString_;
+
+    if (IsBinaryOp()) {
+      CompilerOptions << " -DOPERAND2=";
+      CompilerOptions << (IsScalarOp() ? "InputScalar" : "InputVector2");
+
+      if (IsScalarOp())
+        CompilerOptions << " -DIS_SCALAR_OP=1";
+      else
+        CompilerOptions << " -DIS_BINARY_VECTOR_OP=1";
+
+      CompilerOptions << " -DFUNC=";
+      CompilerOptions << IntrinsicString_;
+    } else { // Unary Op
+      CompilerOptions << " -DFUNC=";
+      CompilerOptions << IntrinsicString_;
+      CompilerOptions << " -DOPERAND2=";
+
+      switch (GetUnaryOpType()) {
+      case LongVectorUnaryOpType_Clamp:
+        CompilerOptions << "ClampArgMinMax";
+        CompilerOptions << " -DFUNC_CLAMP=1";
+        break;
+      case LongVectorUnaryOpType_Initialize:
+        CompilerOptions << " -DFUNC_INITIALIZE=1";
+        break;
+      }
+    }
+
+    return CompilerOptions.str();
+  }
+
+private:
+  std::vector<T> GetInputValueSet(size_t ValueSetIndex) {
+    if (ValueSetIndex == 2 && !IsBinaryOp())
+      VERIFY_FAIL("ValueSetindex==2 is only valid for binary ops.");
+
+    std::wstring InputValueSetName = L"";
+    if (ValueSetIndex == 1)
+      InputValueSetName = InputValueSetName1_;
+    else if (ValueSetIndex == 2)
+      InputValueSetName = InputValueSetName2_;
+    else
+      VERIFY_FAIL("Invalid ValueSetIndex");
+
+    return GetInputValueSetByKey<T>(InputValueSetName);
+  }
+
+  // To be used for the value of -DOPERATOR
+  std::string OperatorString_;
+  // To be used for the value of -DFUNC
+  std::string IntrinsicString_;
+  // Optional, can be used to override shader code.
+  float Tolerance_ = 0.0;
+  LongVectorBinaryOpType BinaryOpType_ = LongVectorBinaryOpType_EnumValueCount;
+  LongVectorUnaryOpType UnaryOpType_ = LongVectorUnaryOpType_EnumValueCount;
+  std::wstring InputValueSetName1_ = L"DefaultInputValueSet1";
+  std::wstring InputValueSetName2_ = L"DefaultInputValueSet2";
+  std::wstring InputArgsArrayName_ = L""; // No default args array
```

Comment:
I'm not sure that underscore suffixes is compatible with the llvm code standards.

Reply:
I had asked co-pilot and it linked me to LLVM documentation saying to add that. But manually navigating to the coding standards documentation and asking co-pilot again doesn't bring it up. Not sure what it showed me...